src/animations.js

```diff
@@ -38,10 +99,31 @@ const ANIMATIONS = {
 }
 
 const CURRENT_ANIMATION_INFO = {
+    karaokeState: null,
     name: null,
     timeScrolledIntoView: null
 }
 
+let FAILED_KARAOKE_UPDATE_ATTEMPTS = 0
+
+
+/**
+ * Even with the correct import order and with the `defer` attribute applied, `updateKaraoke` may not be ready to be
+ * invoked by the time the animation starts.
+ */
+function attemptUpdateKaraoke(karaokeAnimationInfo, animationTime) {
```

Comment:
Just as a note, I tested logging the number of failed attempts just to see what the delay would be. There were only three trials out of a few dozen where it failed at all, and in those three trials it only took between 1 and 3 animation frames before the `updateKaraoke` function was ready.